assets/js/googlesitekit/datastore/user/disconnect.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * core/user Data store: disconnect
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+
+// Actions
+const START_DISCONNECT = 'START_DISCONNECT';
+const FETCH_DISCONNECT = 'FETCH_DISCONNECT';
+const FINISH_DISCONNECT = 'FINISH_DISCONNET';
+const CATCH_FETCH_DISCONNECT = 'CATCH_FETCH_DISCONNECT';
+const RECEIVE_DISCONNECT = 'RECEIVE_DISCONNECT';
+
+export const INITIAL_STATE = {
+	isDisconnecting: false,
+};
+
+export const actions = {
+	*disconnect() {
+		let response, error;
+		yield {
+			payload: {},
+			type: START_DISCONNECT,
+		};
+
+		try {
+			response = yield {
+				payload: {},
+				type: FETCH_DISCONNECT,
+			};
+			yield actions.receiveDisconnect( response );
```

Comment:
I think the receive here is unnecessary as there is no useful response to store here. The API literally just returns `true` right now but the request succeeding is enough by itself for now 😄

Reply:
@tofumatt we don't need `disconnect` state, do we? I suppose it could be used for something but currently disconnect happens with a page reload so not sure if it would really be useful since we keep spinners etc going with local state 🤷

Reply:
I'm personally of the opinion that it's useful to have a state like that—even if right now it's used to force a redirect (`if (isDisconnected) { window.location = '/sign-in'; }` or whatever).

It doesn't really hurt, but it's inline with our other modules so I think is fine.

Reply:
@ryanwelcher please disregard my comments about removing this! 😄

Reply:
Just to be clear, I'll add `disconnected` to the initial state and create a selector for `isDisconnected`?